ElevenLabs/Packages/com.rest.elevenlabs/Runtime/TextToSpeech/TextToSpeechEndpoint.cs

```diff
@@ -264,5 +486,78 @@ private static async Task<string> GetCacheDirectoryAsync(Voice voice)
                 .CreateNewDirectory(nameof(TextToSpeech))
                 .CreateNewDirectory(voice.Id);
         }
+
+        private async Task<string> SaveAudioToCache(byte[] audioBytes, string clipId, Voice voice, OutputFormat outputFormat, CancellationToken cancellationToken)
+        {
+            var audioType = outputFormat.GetAudioType();
+            ValidateAudioType(audioType);
+            
+            var extension = GetFileExtension(audioType);
+            var downloadDirectory = await GetCacheDirectoryAsync(voice);
+            var cachedPath = $"{downloadDirectory}/{clipId}.{extension}";
+
+            if (!File.Exists(cachedPath))
+            {
+                switch (audioType)
+                {
+                    case AudioType.MPEG:
+                        await File.WriteAllBytesAsync(cachedPath, audioBytes, cancellationToken).ConfigureAwait(false);
+                        break;
+                    case AudioType.OGGVORBIS:
+                        var pcmData = PCMEncoder.Decode(audioBytes, PCMFormatSize.SixteenBit);
+                        var frequency = GetFrequencyForPCMFormat(outputFormat);
+                        var oggBytes = await OggEncoder.ConvertToBytesAsync(pcmData, frequency, 1, cancellationToken: cancellationToken).ConfigureAwait(false);
+                        await File.WriteAllBytesAsync(cachedPath, oggBytes, cancellationToken).ConfigureAwait(false);
+                        break;
+                }
+            }
+
+            return cachedPath;
+        }
+        
+        private async Task<VoiceClip> CreateVoiceClipFromFile(string clipId, string text, Voice voice, string cachedPath, OutputFormat outputFormat, CancellationToken cancellationToken)
+        {
+            await Awaiters.UnityMainThread;
+            var audioClip = await Rest.DownloadAudioClipAsync(
+                $"file://{cachedPath}",
+                outputFormat.GetAudioType(),
+                parameters: new RestParameters(debug: EnableDebug),
+                cancellationToken: cancellationToken);
+            return new VoiceClip(clipId, text, voice, audioClip, cachedPath);
+        }
+        
+
+        private void ValidateAudioType(AudioType audioType)
+        {
+            if (audioType is not AudioType.MPEG and not AudioType.OGGVORBIS)
+            {
+                throw new ArgumentOutOfRangeException($"Unsupported {nameof(AudioType)}: {audioType}");
+            }
+        }
+        
+        private void ValidateStreamingFormat(OutputFormat format)
+        {
+            if (format is OutputFormat.MP3_44100_64 or OutputFormat.MP3_44100_96
+                or OutputFormat.MP3_44100_128 or OutputFormat.MP3_44100_192)
```

Comment:
maybe break on `or` for each?
Formatting is a bit strange here. Prob ok if all one line as well.